build_tools/github_actions/post_build_upload.py

```
@@ -210,8 +217,39 @@ def upload_logs_to_s3(run_id: str, amdgpu_family: str, build_dir: Path):
         log(f"[INFO] No index.html found at {log_dir}. Skipping index upload.")
 
 
+def get_manifest_from_build(build_dir: Path):
+    """
+    Look only in the aux-overlay *build* directory.
+    Non-fatal: returns None if not found.
+    """
+    build_path = build_dir / "base" / "aux-overlay" / "build" / "therock_manifest.json"
+
+    if build_path.is_file():
+        return build_path
+
+    return None
+
+
+def upload_manifest_to_s3(run_id: str, amdgpu_family: str, build_dir: Path):
+    """
+    Upload therock_manifest.json to:
+      s3://<bucket>/<external_repo_path><run_id>-<platform>/manifests/<amdgpu_family>/therock_manifest.json
+    """
+    external_repo_path, bucket = get_bucket_info_cached()
+    bucket_uri = f"s3://{bucket}/{external_repo_path}{run_id}-{PLATFORM}"
+
+    manifest = get_manifest_from_build(build_dir)
+    if not manifest:
+        log(f"[WARN] therock_manifest.json not found under {build_dir}")
```

This should throw an exception / be an error